core/tests/protractor_utils/workflow.js

```diff
@@ -264,6 +272,40 @@ var createSkillAndAssignTopic = function(
     0, topicName);
 };
 
+var getThumbnailSource = function(customThumbnailElement) {
+  waitFor.visibilityOf(
+    customThumbnailElement,
+    'Thumbnail element is taking too long to appear.');
+  return customThumbnailElement.getAttribute('src');
+};
+
+var uploadThumbnail = function(thumbnailClickableElement, imgPath) {
```

Comment:
Is there any technical reason though why this wouldn't work for other images? For example, it looks very much like the photo upload code in the preferences page:
https://github.com/oppia/oppia/blob/0e57098315d9f97e32f9134868b2d2fa0d31a619/core/tests/protractor_utils/PreferencesPage.js#L66
Can we create a general utility function to avoid duplicating this code?

Reply:
Ah nice catch! I think we can combine those two and change the function name to a more generic one that suggests a common image upload.

Reply:
Done.